### DISCUSSION TITLE
关于LamdaClass中，对于H(x)的degree补位的疑惑

### DISCUSSION THREAD

**Discussion by 0xxEric** [coset-io/zkp-academy#56] [Q&A]
在[Lambda](https://blog.lambdaclass.com/diving-deep-fri/)中，在计算H(x)时，做了一个补0的操作，我理解它的意图，是补到2^k（比如，这里补到4）
![image](https://github.com/user-attachments/assets/be22bce6-ccda-4d0f-8d42-8f7d4aefb193)

1、我感觉这里处理得不够“general”。它是先得到了多项式：补0前为 $C1(x)=13(x^2+9x+5)$ , $C2(x)=x^3+12x^2+9x+6$ 
也就是说，它是根据它已知计得得到的degree来补。 （那如果算出来是5，就补到8？）
我的疑问是：
我们在实际协议中，是否应该是直接算C(x)在LDE后的Domain上的evaluation，而并不知道它的多项式degree呢？
如果直接把多项式算出来了，那就就已经证明 C(x) is a polynomial了，就不必要再往下做LDT了？ （对比Stark by hand，是在Domain上做点值的有限域运算，而非把多项式coefficent系数算出来）。
实际的协议代码中应该是怎样的呢？不知道这里是否为了教学的简便如此处理的。

2、仍然没有理解到，为何需要补位，如果不补位，直接用现在的 $C(x)=\alpha0 * C1(x) + \alpha1 * C2(x)$ 会带来什么不便之处

**Comment by xrqin** [comment]
我也不是特别理解。我斗胆谈谈我的想法。按照文档的说法，就是如果不补到2的幂次，那么就不能一直砍半。如果只是$`C(x)=\alpha 0 * C 1(x)+\alpha 1 * C 2(x)`$, 那只能砍一次，最后一次肯定不是常数了，因为次数还没降到0。只有补到2的幂次才能砍到常数。

但是我还是不理解的是，verifier怎么保证prover两边都补了，而没有只补一边呢？比如，如果prover只是$`H(x)=C_1(x)\left(\alpha_1 x^{D-D_1}+\beta_1\right)+C_2(x)`$，就是只补了$`C_1`$这边的没补$`C_2`$那边，会发生什么呢？这样子$`H(x)`$也被补到2的幂次了，似乎也是能被砍到常数的呀？

**Comment by Einstellung** [comment]
H的实际计算公式是

$$
H = \sum_k \beta_k^TC_k + \sum_j \beta_j^B B_j
$$

并没有补x那个操作。并没有升degree或许并不关心那个degree一致性的事情，或许用rscode和FRI保证安全性与degree升维无关。然后

> 就是如果不补到2的幂次，那么就不能一直砍半

这种说法是不对的，能否一直保持砍半操作保证来自想要证明的trace长度为 $2^n$ ，或者说order为n。FRI给值的操作全部都是在评估点上去做计算，假设p已经是一个常数项而不是多项式了，我在定义域范围内也能给出值域的。假如说p已经变成常数而FRI砍半还没有砍完，那就继续往下砍好了。
